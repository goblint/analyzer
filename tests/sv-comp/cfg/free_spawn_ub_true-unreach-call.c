extern void __VERIFIER_error() __attribute__ ((__noreturn__));
extern int __VERIFIER_nondet_int();

void foo()
{
    __VERIFIER_error();
}

int main()
{
    void (*p)(void) = &foo;
    free(p); // TODO: free shouldn't spawn
    return 0;
}