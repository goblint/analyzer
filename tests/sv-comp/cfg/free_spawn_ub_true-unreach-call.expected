┌───────────────────────────────────┐
│                 _                 │
└───────────────────────────────────┘
  │
  │ Entry main
  ▼
┌───────────────────────────────────┐
│                 _                 │
└───────────────────────────────────┘
  │
  │ Assign 'p = (void (*)())(& foo)'
  ▼
┌───────────────────────────────────┐
│                 _                 │
└───────────────────────────────────┘
  │
  │ Proc 'free(p)'
  ▼
┌───────────────────────────────────┐
│                 _                 │
└───────────────────────────────────┘
  │
  │ Ret (Some 0, main)
  ▼
┌───────────────────────────────────┐
│                 _                 │
└───────────────────────────────────┘
